src/utils/validators.js

```diff
@@ -0,0 +1,325 @@
+import _isUndefined from 'lodash/isUndefined'
+import _isArray from 'lodash/isArray'
+import _isEmpty from 'lodash/isEmpty'
+import _isObject from 'lodash/isObject'
+
+import { BaseOperation } from '../base/operations/base_operation'
+import { Keypair } from '../base/keypair'
+import { default as xdr } from '../base/generated/xdr_generated'
+
+/**
+ * Validators for encapsulating types checks and throwing errors.
+ *
+ * @example
+ * import { validateUndefined } from './validators'
+ *
+ * function someFunction (someParam) {
+ *   validateUndefined({
+ *     value: someParam,
+ *     fieldName: 'Some field'
+ *   })
+ *
+ *   // do something with someParam
+ * }
+ */
+
+/** Common types validators */
+
+/**
+ * Validate value to be not undefined.
+ *
+ * @param {any} value The value to validate.
+ * @param {string} [fieldName] The name of the field, containing value.
+ *
+ * @throws {TypeError} Value should be defined.
+ */
+export function validateUndefined ({ value, fieldName = '' }) {
+  if (_isUndefined(value)) {
+    throw new TypeError(_composeErrorMessage({
+      value,
+      statement: `should be defined`,
+      fieldName
+    }))
+  }
+}
+
+/**
+ * Validate value to be not NaN (Not a number).
+ *
+ * @param {any} value The value to validate.
+ * @param {string} [fieldName] The name of the field, containing value.
+ *
+ * @throws {TypeError} Value should be not NaN.
+ */
+export function validateNaN ({ value, fieldName = '' }) {
+  if (Number.isNaN(Number(value))) {
+    throw new TypeError(_composeErrorMessage({
+      value,
+      statement: 'should be not NaN',
+      fieldName
+    }))
+  }
+}
+
+/**
+ * Validate value to be an array.
+ *
+ * @param {any} value The value to validate.
+ * @param {string} [fieldName] The name of the field, containing value.
+ * @param {number} [minLength] The minimum array length.
+ *
+ * @throws {TypeError} Value should be an array.
+ * @throws {Error} Array should have at least minimum length.
+ */
+export function validateArray ({ value, fieldName = '', minLength }) {
+  if (!_isArray(value)) {
+    throw new TypeError(_composeTypeErrorMessage({
+      value,
+      expectedType: 'array',
+      fieldName
+    }))
+  }
+
+  if (value.length < minLength) {
+    throw new Error(_composeErrorMessage({
+      value,
+      statement: `must have minimum length of ${minLength}`,
+      fieldName
+    }))
+  }
+}
+
+/**
+ * Validate value to be a string.
+ *
+ * @param {any} value The value to validate.
+ * @param {string} [fieldName] The name of the field, containing value.
+ * @param {number} [minLength] The minimum string length.
+ * @param {number} [maxLength] The maximum string length.
+ *
+ * @throws {TypeError} Value should be a valid string.
+ */
+export function validateString ({ value, fieldName = '', minLength, maxLength }) {
```

Comment:
what if I won't pass `minLength` and `maxLength` and just want to validate the value to be a string?

Reply:
`minLength` & `maxLength` params are optional. If some of them are not passed, the validator ignores them.

Reply:
Is it necessary? And there's a problem here. We can set `minLength = 0`, but what about `maxLength`? Should it be `maxLength = Infinity` or some finite string length? But actually maximum string length depends on the browser and cannot be preset definitely.

Also, there's a problem with error messages. IMO `some_field must be a valid string with params: {}` has better readability than `some_field must be a valid string with params: {"minLength":0,"maxLength":4294967295}` or smth like this.